src/Libraries/Microsoft.Extensions.Caching.Hybrid/HybridCacheServiceExtensions.cs

```diff
@@ -41,4 +44,36 @@ public static IHybridCacheBuilder AddHybridCache(this IServiceCollection service
         services.TryAddSingleton<HybridCache, DefaultHybridCache>();
         return new HybridCacheBuilder(services);
     }
+
+    /// <summary>
+    /// Adds support for keyed multi-tier caching services.
+    /// </summary>
+    /// <returns>A builder instance that allows further configuration of the <see cref="HybridCache"/> system.</returns>
+    [System.Diagnostics.CodeAnalysis.Experimental("CACHE001")]
+    public static IHybridCacheBuilder AddHybridCache(this IServiceCollection services, string name, Action<HybridCacheOptions> setupAction)
+    {
+        _ = Throw.IfNull(services);
+        _ = Throw.IfNull(name);
+        _ = Throw.IfNull(setupAction);
+
+        // Register options for this key
+        _ = services.AddOptions<HybridCacheOptions>(name).Configure(setupAction);
+
+        // Register keyed in-memory cache
+        services.TryAddKeyedSingleton<IMemoryCache, MemoryCache>(name);
+
+        // Register keyed HybridCache
+        services.TryAddKeyedSingleton<HybridCache>(name, (sp, _) =>
+            new DefaultHybridCache(
+                Options.Options.Create(sp.GetRequiredService<IOptionsMonitor<HybridCacheOptions>>().Get(name)),
+                sp,
+                sp.GetKeyedService<IMemoryCache>(name),
+                sp.GetKeyedService<IDistributedCache>(name)));
```

Comment:
Requiring that the `name` for this hybrid cache must match the name of both the `IMemoryCache` and `IDistributedCache` keyed services is somewhat limiting. I'm exploring a different approach here that would support each of the names being distinct from each other.